<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Обработка ошибок 79.01</title>
</head>

<body>
    <script>
        let promise = new Promise((resolve, reject) => {
            setTimeout(() => {
                // resolve(5);
                reject ('er')
             
            }, 3000)
        })

        promise.then(
            (result) => {
                return new Promise((resolve, reject) => {
                    setTimeout(() => {
                        resolve(result)
                    }, 3000)
                })
            }
        ).then(
            result => {
                return new Promise((resolve, reject) => {
                    setTimeout(() => {
                        resolve(result ** 2)
                        reject('error');
                    }, 3000)
                })
            }
        ).then(
            result => console.log(result ** 2)
        ).catch( err => console.log(err))
           
        
    </script>

</body>

</html>